hw/ip/aes/README.md

```diff
@@ -33,13 +33,18 @@ This AES unit targets medium performance (16 parallel S-Boxes, \~1 cycle per rou
 High-speed, single-cycle operation for high-bandwidth data streaming is not required.
 
 Cipher modes other than ECB, CBC, CFB, OFB and CTR are beyond this version of the AES unit but might be supported in future versions.
+Galois/Counter Mode (GCM) can be implemented by leveraging [Ibex](../rv_core_ibex/README.md) for the GHASH operation as demonstrated in [OpenTitan's library of cryptographic implementations](https://github.com/lowRISC/opentitan/tree/master/sw/device/lib/crypto).
 
 
 ## Description
 
 The AES unit is a cryptographic accelerator that accepts requests from the processor to encrypt or decrypt 16B blocks of data.
 It supports AES-128/192/256 in Electronic Codebook (ECB) mode, Cipher Block Chaining (CBC) mode, Cipher Feedback (CFB) mode (fixed data segment size of 128 bits, i.e., CFB-128), Output Feedback (OFB) mode and Counter (CTR) mode.
 For more information on these cipher modes, refer to [Recommendation for Block Cipher Modes of Operation](https://nvlpubs.nist.gov/nistpubs/Legacy/SP/nistspecialpublication800-38a.pdf).
+Galois/Counter Mode (GCM) can be implemented using [Ibex](../rv_core_ibex/README.md) for the GHASH operation as demonstrated in [OpenTitan's library of cryptographic implementations](https://github.com/lowRISC/opentitan/tree/master/sw/device/lib/crypto).
+To improve the performance of GCM, instructions of the [RISC-V Bit-Manipulation Extension of Ibex](https://ibex-core.readthedocs.io/en/latest/03_reference/instruction_decode_execute.html#arithmetic-logic-unit-alu) can be leveraged.
```

Comment:
The current GCM implementation doesn't use the bit-manipulation extension because I've never been able to get it to work. Even after running `util/get-toolchain.py --upgrade`, if I try to use the `grev` instruction I get compiler errors:
```
sw/device/lib/crypto/impl/aes_gcm/ghash.c:138:7: error: instruction requires the following: 'Zbp' (Permutation 'Zb' Instructions)
      "grev %[dest], %[word], 0x18;"
```
Am I doing something wrong here? Last I heard from @luismarques we were waiting on a toolchain upgrade for OT, so I've been perioidically checking since, but the errors have never gone away.

Reply:
Hi @jadephilipoom. Sorry for the delay in replying, I missed your comment. The toolchain supports all the bit manipulation extensions but we're only enabling by default the ratified extensions Zba, Zbb, Zbc, and Zbs. For the other ones, I recommend enabling them only where they are strictly needed, by using the following inline assembly sequence:

```
.option push
.option arch, +zbp0p93
... use grev etc.
.option pop
```

See for instance `sw/device/silicon_creator/lib/crc32.c`. I hope that clears it up. Let me know if you run into any problems, feel free to contact me directly over Slack or email.

BTW, we're no longer using `util/get-toolchain.py` for getting the toolchain that Bazel uses to build the software. It's managed using Bazel / CRT. Since the Bazel toolchain install isn't very convenient for other uses it might make sense to keep that script but it has limited utility.

Reply:
That works, thanks! I didn't know I needed to explicitly enable that extension before using it, that's what I was missing :+1: